std/process.d

```diff
@@ -4194,7 +4194,8 @@ else version (Posix)
             //args[0] = "x-www-browser".ptr;  // doesn't work on some systems
             args[0] = "xdg-open".ptr;
 
-        args[1] = url.tempCString();
+        const buffer = url.tempCString(); // Retain buffer until end of scope
+        args[1] = buffer;
```

Comment:
The `version(OSX)` above is almost identical, except that it already used a local for the buffer and uses `open` instead of `xdg-open`. If the 2 extra array elements for Mac are superfluous, the 2 blocks could be unified.

Reply:
`execvp` expects a `null` pointer as terminator for the array of arguments, so the two superfluous elements shouldn't matter.

Reply:
Added a commit to merge both implementations (to not duplicate the additional check for `StdUnittest`)

Reply:
Thanks!

Reply:
Would `-dip1000` have captured this?